res_partner_stock_picking/models/stock_picking.py

```diff
@@ -0,0 +1,14 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from odoo import models
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _pre_action_done_hook(self):
+        for picking in self:
+            never_parcial_recepcion = picking.partner_id.create_partial_reception
+            if never_parcial_recepcion == 'never':
+                self.with_context(skip_backorder=True)
+            else:
+                super()._pre_action_done_hook()
```

Comment:
This code is wrong because of two problems:
* For `never` case `super()` method is never called. You only change context, and then exit.
* When else clause is executed you're calling original method for every picking in self, not for the current picking in the loop, because `super()` is a shortcut of `super(StockPicking, self)`, e.g. called for the whole records.